platform/security/encryption/platform-security-encryption-impl/src/main/java/ddf/security/encryption/impl/EncryptionServiceImpl.java

```diff
@@ -28,54 +42,75 @@
 
   private static final Pattern ENC_PATTERN = Pattern.compile("^ENC\\((.*)\\)$");
   private static final String ENC_TEMPLATE = "ENC(%s)";
+  private static final String KEYSET_FILE_NAME = "keyset.json";
+  private static final byte[] ASSOCIATED_DATA = "associated.data".getBytes();
 
-  private final Crypter crypter;
+  @VisibleForTesting KeysetHandle keysetHandle;
+  private Aead aead;
 
   public EncryptionServiceImpl() {
     String passwordDirectory = System.getProperty("ddf.etc").concat("/certs");
+    String keysetLocation = Paths.get(passwordDirectory, KEYSET_FILE_NAME).toString();
```

Comment:
✏️ This ctor is similarly long and doing an awful lot inline. Could be broken up.

Then again...I'm wondering whether the Tink library could not be abstracted away from this and the `EncryptingPersistenceManager` into a new library jar that could be embedded. That jar would expose the bare minimum of methods - `init`, `encrypt`, and `decrypt` should be it, I think - and all of this work would be hidden from the users of it. Then if we needed to swap in a new library in the future to replace Tink, it would require changes in only the implementation of that single library.

I think this is the right approach here.

Reply:
👍 I like this solution a lot. Plus it (probably) would solve the weird bundle loading issues that this PR currently has.